IGraphics/IGraphics.h

```diff
@@ -970,6 +970,17 @@ class IGraphics
   /** Get the app group ID on macOS and iOS, returns emtpy string on other OSs */
   virtual const char* GetAppGroupID() const { return ""; }
 
+  // An RAII helper to manage the IGraphics GL context
+  class ScopedGLContext
+  {
+  public:
+    ScopedGLContext(IGraphics* pThis)
+    : mThis(pThis) { mThis->ActivateGLContext(); }
+    ~ScopedGLContext() { mThis->DeactivateGLContext(); }
+  private:
+    IGraphics* mThis;
+  };
```

Comment:
This looks like a good idea, but I'd avoid using "this" as a member/parameter naming, as it could also be used from a Mac view correctly.

As this object also makes no sense if there is no IGraphics I'd suggest it should be stored by reference to the object, rather than by pointer.

Reply:
I tweaked